examples/vercel-ai-chatbot/app/api/chat/route.ts

```diff
@@ -22,42 +31,48 @@ export async function POST(req: Request) {
     })
   }
 
-  if (previewToken) {
-    openai.apiKey = previewToken
-  }
-
-  const res = await openai.chat.completions.create({
-    model: 'gpt-3.5-turbo',
-    messages,
-    temperature: 0.7,
-    stream: true
+  const res = await vellum.executeWorkflowStream({
+    workflowDeploymentName: 'vercel-chatbot-demo',
+    releaseTag: 'production',
+    inputs: [
+      {
+        type: 'CHAT_HISTORY',
+        value: messages.map(m => ({
+          role: m.role.toUpperCase() as ChatMessageRole,
+          text: m.content
+        })),
+        name: 'messages'
+      }
+    ]
   })
 
-  const stream = OpenAIStream(res, {
-    async onCompletion(completion) {
-      const title = json.messages[0].content.substring(0, 100)
-      const id = json.id ?? nanoid()
-      const createdAt = Date.now()
-      const path = `/chat/${id}`
-      const payload = {
-        id,
-        title,
-        userId,
-        createdAt,
-        path,
-        messages: [
-          ...messages,
-          {
-            content: completion,
-            role: 'assistant'
+  const stream = new ReadableStream({
+    async pull(controller) {
+      for await (const event of res) {
+        if (event.type !== 'WORKFLOW') {
+          continue
+        }
+        if (event.data.state === 'REJECTED') {
+          controller.enqueue(JSON.stringify(event.data.error!))
+          controller.close()
+          break
+        }
+        if (event.data.state === 'INITIATED') {
+          continue
+        }
+        if (event.data.state === 'STREAMING') {
+          const output = event.data.output!
+          if (output.state === 'STREAMING') {
+            controller.enqueue(output.delta)
           }
-        ]
+          // if (output.state === 'FULFILLED' && output.type === 'FUNCTION_CALL') {
+          //   controller.enqueue(output.value)
+          // }
+        }
+        if (event.data.state === 'FULFILLED') {
+          controller.close()
+        }
       }
-      await kv.hmset(`chat:${id}`, payload)
-      await kv.zadd(`user:chat:${userId}`, {
```

Comment:
How tied to Vercel should I make this template?
- Should I support their KV store by making one in our account?
- Should I get Github oauth working by integrating our Vellum Github app?
- Should I get a deployed Vercel url accessible?

I _think_ we have a Vercel account, right?

Reply:
I connected the kv stuff so you can bring this back. The kv store is just redis so we can always use a non-Vercel one later if we want. I just created one in Vercel for now.

Reply:
Legendary